front/src/applications/stdcm/components/StdcmResults/SimulationReportSheet.tsx

```diff
@@ -305,7 +311,13 @@ const SimulationReportSheet = ({
                 const prevStep = operationalPointsList[index - 1];
                 const isViaInSimulationPath = stdcmData.simulationPathSteps
                   .slice(1, -1)
-                  .some((pathStep) => pathStep.name === step.name && pathStep.ch === step.ch);
+                  .some(
+                    ({ location }) =>
+                      location &&
+                      location.name === step.name &&
+                      'uic' in location &&
+                      location.secondary_code === step.ch
```

Comment:
Would be nicer to use `getSecondaryCode()` here instead.

Reply:
Done